### PR TITLE
Refactor flag handling

### PR DESCRIPTION
**Pull request by mkmik**
The run method was too long and it was impossible to figure out what each parameter meant.
Furthermore a few flags were still accessed via global variables making it hard to reason about
and to test properly.

This is the first pass that just mechanically puts all the flag variables into a struct.

### CODE REVIEW

File: cmd/kubeseal/main.go [hunk 1]
```diff
@@ -78,14 +58,60 @@ var (
...
+	fs.StringVar(&f.mergeInto, "merge-into", "", "Merge items from secret into an existing sealed secret file, updating the file in-place instead of writing to stdout.")
+	fs.BoolVar(&f.raw, "raw", false, "Encrypt a raw value passed via the --from-* flags instead of the whole secret object")
+	fs.StringVar(&f.secretName, "name", "", "Name of the sealed secret (required with --raw and default (strict) scope)")
+	fs.StringSliceVar(&f.fromFile, "from-file", nil, "(only with --raw) Secret items can be sourced from files. Pro-tip: you can use /dev/stdin to read pipe input. This flag tries to follow the same syntax as in kubectl")
+
+	fs.Var(&f.sealingScope, "scope", "Set the scope of the sealed secret: strict, namespace-wide, cluster-wide (defaults to strict). Mandatory for --raw, otherwise the 'sealedsecrets.bitnami.com/cluster-wide' and 'sealedsecrets.bitnami.com/namespace-wide' annotations on the input secret can be used to select the scope.")
+	fs.BoolVar(&f.reEncrypt, "rotate", false, "")
+	fs.BoolVar(&f.reEncrypt, "re-encrypt", false, "Re-encrypt the given sealed secret to use the latest cluster key.")
 	flag.CommandLine.MarkDeprecated("rotate", "please use --re-encrypt instead")
```
Comment (aruiz14) [inline cmd/kubeseal/main.go:106]: This should be `fs.MarkDeprecated` instead, right?

Comment (mkmik) [reply cmd/kubeseal/main.go:106]: Good catch, my clever regexp based refactoring was not so clever in retrospect

**Comment by juan131** [comment]
Let's put this PR on-hold until [the discussion around refactoring kubeseal](https://github.com/bitnami-labs/sealed-secrets/issues/717) is done.

**Comment by josvazg** [comment]
I sent this other PR recently, https://github.com/bitnami-labs/sealed-secrets/pull/901 as I felt I had to fix those global variables before tackling the run and flags mess. But was planning on going this route next.

I would like to push for this PR to move forward as soon as possible.

**Comment by josvazg** [comment]
Due to the age of this PR and, the rebase and resume work has moved to this other branch and PR: #920 **Rebase of Refactor flag handling**

Closing this one and moving review to the new PR instead.

**Pull request closed without merging**